app/controllers/change_history_controller.rb

```diff
@@ -0,0 +1,41 @@
+class ChangeHistoryController < ApplicationController
```

Comment:
Just to point out the naming unrailsiness, in case you've had additional thoughts since our chat.